commit 44692f28d859728fb919a5d736ada82606196f75
Author: Johannes Brechtmann <johannes@jnbr.me>
Date:   Wed Jun 26 22:51:41 2019 +0200

    bin/xbps-install: update_pkg return EEXIST if package is up to date
    
    makes 58509996aabea52ffc40e5e01c9eb00730c6cdcc work for
    xbps-install -u <pkg>

diff --git a/bin/xbps-install/transaction.c b/bin/xbps-install/transaction.c
index cdcc92d7..fc5538bb 100644
--- bin/xbps-install/transaction.c
+++ bin/xbps-install/transaction.c
@@ -286,7 +286,7 @@ update_pkg(struct xbps_handle *xhp, const char *pkgname)
 	rv = xbps_transaction_update_pkg(xhp, pkgname);
 	if (rv == EEXIST) {
 		printf("Package '%s' is up to date.\n", pkgname);
-		return 0;
+		return EEXIST;
 	} else if (rv == ENOENT)
 		fprintf(stderr, "Package '%s' not found in "
 		    "repository pool.\n", pkgname);
